### PR TITLE
Implement and test `n4::place::trans(G4Transform3D*)`

### LINKED ISSUE #72: Add placement method that accepts a `G4Transform3D`
Name?

+ `trans'
+ `trans_3d`
+ `at_rot`

### PR DESCRIPTION
**Pull request by gonzaponte**
Closes #72.

Perhaps, like with `rotate`/`rot` we can have `transform`/`trans`?

### CODE REVIEW

**Comment by jacg** [comment]
> Perhaps, like with `rotate`/`rot` we can have `transform`/`trans`?

Slightly cryptic, but probably OK. Go for it.

File: nain4/test/test-nain4.cc [hunk 1]
```diff
@@ -1061,6 +1063,19 @@ TEST_CASE("nain place", "[nain][place]") {
     CHECK(* rot_xyz    -> GetObjectRotation() == rotmat(angle, angle, angle));
   }
 
+  SECTION("transforms") {
+    auto water  = nain4::material("G4_WATER");
+    auto box    = nain4::volume<G4Box>("box", water, 0.3*m, 0.2*m, 0.1*m);
```
Comment (jacg) [inline nain4/test/test-nain4.cc:1068]: Any reason to use `volume<G4Box>` instead of `n4::box`?

(Same thing happened in the last PR.)

**Comment by jacg** [comment]
Please make sure that the placement docs (I've pushed the source, and rebased this on top of it) list any new methods, in this PR.

File: nain4/test/test-nain4.cc (hunk 1 above)
Comment (gonzaponte) [reply nain4/test/test-nain4.cc:1068]: No, just a copy-paste from another section. I will change them all.

**Pull request merged**